src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,31 @@
+package duke;
+
+/**
+ * Exception class
+ * Print various messages to users due to different error situations
+ */
+public class DukeException extends Exception {
+    DukeException(ExceptionType e) {
+        switch (e) {
```

Comment:
I like you have used switch case to tackle different exception, but i would recommend to extend the different exceptions out into different exception class. That would be easier to print out exception message when implementing GUI.

Reply:
Likewise, I feel the exception handling could be further split into individual classes, but most exceptions are handled so well done on that.